src/engines/ngspice-watcher.c

```diff
@@ -302,7 +302,7 @@ enum NGSPICE_WATCHER_RETURN_VALUE {
 static enum NGSPICE_WATCHER_RETURN_VALUE
 ngspice_watcher_watch_ngspice_resources (GPid pid, gint status, NgSpiceWatcherWatchNgSpiceResources *resources) {
 	GThread *worker = resources->worker;
-	GThread *saver = resources->saver;
+	GThread **saver = resources->saver;
```

Comment:
add a ref each, and unref, this makes sure the thread is always valid (even if the side conditions already might make sure that this is the case)

Reply:
Not sure what I meant with that :shrug: